src/cardreader.cpp

```diff
@@ -392,20 +393,24 @@ void CardReader::HandlerFunction(void* arg) {
 
         if (!keep_watching) {
             reader->m_state = 2;
+            if (result == SCARD_E_NO_READERS_AVAILABLE) { // It's not an error, we send reader removed event (do_exit==true)
```

Comment:
We can move this to the `HandleReaderStatusChange` function in the same way we do in `pcsclite.cpp`